modules/library/referencing/src/main/java/org/geotools/referencing/factory/epsg/EnsembleDefinition.java

```diff
@@ -0,0 +1,165 @@
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.referencing.factory.epsg;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.geotools.metadata.iso.citation.Citations;
+import org.geotools.referencing.NamedIdentifier;
+
+/**
+ * This class contains Ensemble datum definitions and mapping to connect an Ensemble datum to a specific version of the
+ * datum.
+ */
+class EnsembleDefinition {
+    String name;
+    NamedIdentifier nameIdentifier;
+    String code;
+    String ellipsoidCode;
+    String primeMeridianCode;
+    String identifierAuthority;
+
+    // The selected datum code is the one matching the INFORMATION_SOURCE
+    // of the datum used by the previous version of EPSG DB.
+    String datumCode;
+    boolean vertical;
+
+    public EnsembleDefinition(
+            String name,
+            NamedIdentifier nameIdentifier,
+            String code,
+            String ellipsoidCode,
+            String primeMeridianCode,
+            String identifierAuthority,
+            String datumCode,
+            boolean vertical) {
+        this.name = name;
+        this.nameIdentifier = nameIdentifier;
+        this.code = code;
+        this.ellipsoidCode = ellipsoidCode;
+        this.primeMeridianCode = primeMeridianCode;
+        this.identifierAuthority = identifierAuthority;
+        this.datumCode = datumCode;
+        this.vertical = vertical;
+    }
+
+    public static boolean hasId(String identifier) {
+        return ENSEMBLE_MAP_BY_NAME.containsKey(identifier);
+    }
+
+    public static boolean hasCode(String code) {
+        return ENSEMBLE_MAP_BY_CODE.containsKey(code);
+    }
+
+    public static EnsembleDefinition getEnsemble(String epsg) {
+        EnsembleDefinition result = null;
+        if (ENSEMBLE_MAP_BY_CODE.containsKey(epsg)) {
+            result = ENSEMBLE_MAP_BY_CODE.get(epsg);
+        }
+        return result;
+    }
+
+    public static boolean isExcludedOperation(String operationCode) {
+        return EXCLUDED_OPERATIONS.contains(operationCode);
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public NamedIdentifier getNameIdentifier() {
+        return nameIdentifier;
+    }
+
+    public String getCode() {
+        return code;
+    }
+
+    public String getEllipsoidCode() {
+        return ellipsoidCode;
+    }
+
+    public String getPrimeMeridianCode() {
+        return primeMeridianCode;
+    }
+
+    public String getIdentifierAuthority() {
+        return identifierAuthority;
+    }
+
+    public String getDatumCode() {
+        return datumCode;
+    }
+
+    public boolean isVertical() {
+        return vertical;
+    }
+
+    static final Map<String, EnsembleDefinition> ENSEMBLE_MAP_BY_NAME;
+    static final Map<String, EnsembleDefinition> ENSEMBLE_MAP_BY_CODE;
+    static final Set<String> EXCLUDED_OPERATIONS;
```

Comment:
EXCLUDED_OPERATIONS... how is it related to the ensemble?

Reply:
I have renamed the class afterwards to dedicate it to EnsembleDefinition.
I will move it to the CoordinateOperationSet.